### PR TITLE
USSD prevent multiple registrations

### PR DESCRIPTION
**Pull request by NicolaasDieBaas**

### CODE REVIEW

**Comment by NicolaasDieBaas** [comment]
@gsvr plz review

File: test/ussd_registration.test.js [hunk 1]
```diff
@@ -249,6 +249,87 @@ describe("Mama Nigeria App", function() {
...
+                        state: 'state_msisdn_already_registered',
+                        reply: [
+                            "Sorry, this number is already registered. They must opt-out before registering again.",
+                            "1. Try a different number",
+                            "2. Choose a different receiver",
+                            "3. Exit"
+                        ].join('\n')
+                    })
+                    .check(function(api) {
+                        var fixt78 = api.http.fixtures.fixtures[78];
```
Comment (gsvr) [inline test/ussd_registration.test.js]: Please use the other fixture usage test - this was actually an old prototype.

File: test/ussd_registration.test.js [hunk 2]
```diff
@@ -249,6 +249,87 @@ describe("Mama Nigeria App", function() {
...
+                        , '12345'  // state_auth_code - personnel code
+                        , '2' // state_msg_receiver - mother_only
+                        , '09097777777'  // state_msisdn
+                        , '3' // state_end_msisdn - exit
+                    )
+                    .check.interaction({
+                        state: 'state_end_msisdn',
+                        reply: "Thank you for using the Hello Mama service."
+                    })
+                    .run();
```
Comment (gsvr) [inline test/ussd_registration.test.js:305]: Add the fixture tests for all the new tests, please.

File: test/ussd_registration.test.js [hunk 3]
```diff
@@ -393,8 +474,9 @@ describe("Mama Nigeria App", function() {
                     .inputs(
                       {session_event: 'new'}  // dial in
                       , '12345'   // state_auth_code - personnel code
-                      , '6' // state_msg_receiver - friend_only
-                      , '09092222222'  // state_msisdn
+                      , '1' // state_msg_receiver - mother and father
+                      , '09092222222'  // state_msisdn_mother
+                      , '09091111111'  // state_msisdn_household
```
Comment (gsvr) [inline test/ussd_registration.test.js:494]: why was this changed?

File: test/ussd_registration.test.js [hunk 4]
```diff
@@ -256,7 +337,7 @@ describe("Mama Nigeria App", function() {
                         {session_event: 'new'}  // dial in
                         , '12345'   // state_auth_code - personnel code
                         , '2' // state_msg_receiver - mother_only
-                        , '09091111111'  // state_msisdn
+                        , '09092222222'  // state_msisdn
```
Comment (gsvr) [inline test/ussd_registration.test.js]: why was this changed?

File: test/ussd_registration.test.js (hunk 3 above)
Comment (NicolaasDieBaas) [inline test/ussd_registration.test.js:494]: i noticed a lot of 'friend_only'/'1' choices for state_msg_receiver so chose to change it and temporarily wanted to reach state_msg_language by a different branch of flow... i could change it back..?

File: test/ussd_registration.test.js (hunk 4 above)
Comment (NicolaasDieBaas) [inline test/ussd_registration.test.js]: can't remember... changing it back

**Comment by NicolaasDieBaas** [comment]
@gsvr another review plz

File: test/ussd_registration.test.js (hunk 3 above)
Comment (gsvr) [inline test/ussd_registration.test.js:494]: ¯\_(ツ)_/¯ your call

File: test/ussd_registration.test.js [hunk 5]
```diff
@@ -249,6 +249,101 @@ describe("Mama Nigeria App", function() {
...
+                    .check(function(api) {
+                        var expected_used = [1,6,78];
+                        var fixts = api.http.fixtures.fixtures;
+                        var fixts_used = [];
+                        fixts.forEach(function(f, i) {
+                            f.uses > 0 ? fixts_used.push(i) : null;
+                        });
+                        assert.deepEqual(fixts_used, expected_used);
+                    })
+                    .run();
```
Comment (gsvr) [inline test/ussd_registration.test.js:305]: add `.check.reply.ends_session()`

File: test/ussd_registration.test.js [hunk 6]
```diff
@@ -249,6 +249,101 @@ describe("Mama Nigeria App", function() {
...
+            });
+            it("to state_msisdn (from state_msisdn_already_registered)", function() {
+                return tester
+                    .setup.user.addr('08080020002')
+                    .inputs(
+                        {session_event: 'new'}  // dial in
+                        , '12345'  // state_auth_code - personnel code
+                        , '2' // state_msg_receiver - mother_only
+                        , '09097777777'  // state_msisdn
+                        , '1' // state_end_msisdn - exit
```
Comment (gsvr) [inline test/ussd_registration.test.js]: Comment looks wrong here

File: test/ussd_registration.test.js [hunk 7]
```diff
@@ -249,6 +249,101 @@ describe("Mama Nigeria App", function() {
...
+            });
+            it("to state_msg_receiver (from state_msisdn_already_registered)", function() {
+                return tester
+                    .setup.user.addr('08080020002')
+                    .inputs(
+                        {session_event: 'new'}  // dial in
+                        , '12345'  // state_auth_code - personnel code
+                        , '2' // state_msg_receiver - mother_only
+                        , '09097777777'  // state_msisdn
+                        , '2' // state_end_msisdn - exit
```
Comment (gsvr) [inline test/ussd_registration.test.js]: Comment looks wrong here.

**Comment by gsvr** [comment]
Just fix the comments, merge develop, and then :+1: when travis is happy.
